Backend/ClientActions.php

```diff
@@ -41,10 +41,12 @@ public function getZonesReturnMagentoZone() {
         $response = $this->api->callAPI($this->request);
         if($this->api->responseOk($response)) {
             $magentoZone = null;
+            $bestMatch = strlen($magentoDomainName);
             foreach ($response['result'] as $zone) {
-                if (strstr($magentoDomainName, $zone['name'])) {
+                $firstOccurrence = strpos($magentoDomainName, $zone['name']);
+                if ($firstOccurrence !== false && $firstOccurrence < $bestMatch) {
+                    $bestMatch = $firstOccurrence;
```

Comment:
I might be misunderstanding you but I don't think this is a valid use case.  `$magentoDomainName` is the domain we get from Magento, it will always be greater than or equal to the length of the CloudFlare domain.   For your example to be valid (I think) it would have to be:

```
$magentoDomainName = 'verylongsub.domain.com';
$zone = 'domain.com';
```

Which would still work correctly right?